rust/sedona-schema/src/datatypes.rs

```diff
@@ -333,6 +365,1241 @@ fn deserialize_edges(edges: &Value) -> Result<Edges> {
+/// Utilizing nested structs and lists to represent raster metadata and bands.
+#[derive(Debug, PartialEq, Clone)]
+pub struct RasterSchema;
+impl RasterSchema {
+    // Raster schema:
+    pub fn fields() -> Fields {
+        Fields::from(vec![
+            Field::new(column::METADATA, Self::metadata_type(), false),
+            Field::new(column::CRS, Self::crs_type(), true),
+            Field::new(column::BBOX, Self::bounding_box_type(), true),
+            Field::new(column::BANDS, Self::bands_type(), true),
+        ])
+    }
+
+    /// Raster metadata schema
+    pub fn metadata_type() -> DataType {
+        DataType::Struct(Fields::from(vec![
+            // Raster dimensions
+            Field::new(column::WIDTH, DataType::UInt64, false),
+            Field::new(column::HEIGHT, DataType::UInt64, false),
+            // Geospatial transformation parameters
+            Field::new(column::UPPERLEFT_X, DataType::Float64, false),
+            Field::new(column::UPPERLEFT_Y, DataType::Float64, false),
+            Field::new(column::SCALE_X, DataType::Float64, false),
+            Field::new(column::SCALE_Y, DataType::Float64, false),
+            Field::new(column::SKEW_X, DataType::Float64, false),
+            Field::new(column::SKEW_Y, DataType::Float64, false),
+        ]))
+    }
+
+    /// Bounding box schema
+    pub fn bounding_box_type() -> DataType {
+        DataType::Struct(Fields::from(vec![
+            Field::new(column::MIN_X, DataType::Float64, false),
+            Field::new(column::MIN_Y, DataType::Float64, false),
+            Field::new(column::MAX_X, DataType::Float64, false),
+            Field::new(column::MAX_Y, DataType::Float64, false),
+        ]))
+    }
+
+    /// Bands list schema
+    pub fn bands_type() -> DataType {
+        DataType::List(FieldRef::new(Field::new(
+            column::BAND,
+            Self::band_type(),
+            false,
+        )))
+    }
+
+    /// Individual band schema
+    pub fn band_type() -> DataType {
+        DataType::Struct(Fields::from(vec![
+            Field::new(column::METADATA, Self::band_metadata_type(), false),
+            Field::new(column::DATA, Self::band_data_type(), false),
+        ]))
+    }
+
+    /// Band metadata schema
+    pub fn band_metadata_type() -> DataType {
+        DataType::Struct(Fields::from(vec![
+            Field::new(column::NODATAVALUE, DataType::Binary, true), // Allow null nodata values
+            Field::new(column::STORAGE_TYPE, DataType::UInt32, false),
+            Field::new(column::DATATYPE, DataType::UInt32, false),
+            // OutDb reference fields - only used when storage_type == OutDbRef
+            Field::new(column::OUTDB_URL, DataType::Utf8, true),
+            Field::new(column::OUTDB_BAND_ID, DataType::UInt32, true),
+        ]))
+    }
+
+    /// Band data schema (single binary blob)
+    pub fn band_data_type() -> DataType {
+        DataType::Binary // consider switching to BinaryView
```

Comment:
I believe that `BinaryView` is a better option. Growing the binary array when appending in-db band data could be less painful.